pkg/api/api.go

```diff
@@ -194,9 +196,53 @@ func (s *server) newTracingHandler(spanName string) func(h http.Handler) http.Ha
 	}
 }
 
+// checkOrigin returns true if the origin is not set or is equal to the request host.
+func (s *server) checkOrigin(r *http.Request) bool {
+	origin := r.Header["Origin"]
+	if len(origin) == 0 {
+		return true
+	}
+	u, err := url.Parse(origin[0])
+	if err != nil {
+		return false
+	}
+	hosts := append(s.CORSAllowedOrigins, r.Host)
+	for _, v := range hosts {
+		if equalASCIIFold(u.Host, v) || v == "*" {
```

Comment:
This check is validating only the Host part of the URL, ignoring the Schema part. This is a backward incompatible change as s.CORSAllowedOrigins were required to be a full URL and now just a Host part of it on order to pass this condition.

Reply:
i see. but then we can't append `r.Host`, can we? since it only hostname+port, but does not include the schema

Reply:
Correct, we don't even need to urlparse origin[0], just to append origin[0] as it is.

Reply:
i added some solution, let me know what you think. it could be that having a really correct solution which is totally by-the-book is not realistic right now, as there are some details here and we need to document how different client set the header

Reply:
The only thing is having the correct scheme. (comment below)

Reply:
i merged it, although i'm not sure how this behave behind a load balancer that rewrites an HTTPS request into a regular HTTP request that goes into the actual bee node (since bee does not offer HTTPS)

Reply:
Correct, we do not have tls connections, but if someone would add them, then this code will not be a surprise to be handled, it will work. For proxies, the configuration would need to include the exact origin in CORSAllowedOrigins with https scheme. We are already setting `*` on gateways so it should work there.